### DISCUSSION TITLE
Change caret color by mode

### DISCUSSION THREAD

**Discussion by lucasbalestero** [JetBrains/ideavim#499] [Ideas]
I'd like to set a caret color for each mode. Is there already a way of doing this? e.g. red caret when in insert mode, green when in visual, etc...

**Comment by citizenmatt** [comment]
This isn't supported right now. The `'guicursor'` support will change the shape of the cursor based on mode, but not the colour.

(The `'guicursor'` setting does allow specifying a highlight group as part of the cursor attributes, so if we supported highlight groups, this could be a way of making it work. But we don't support highlight groups - that would be opening up a can of worms between Vim and IntelliJ syntax highlighting and how to configure the colours used…)

**Reply by lucasbalestero** [reply]
Hmmm, yeah makes sense. I agree it could start getting a little messy... but in the other hand it would open up for some pretty cool possibilities. Either way, thanks for the reply!
